packages/api/src/external/commonwell/patient.ts

```diff
@@ -152,6 +152,77 @@ export async function create(
   }
 }
 
+export async function retryLinking(patient: Patient, facilityId: string): Promise<boolean> {
+  let commonWell: CommonWellAPI | undefined;
+  try {
+    const { debug } = Util.out(`CW retry linking - M patientId ${patient.id}`);
+
+    const { organization, facility } = await getPatientData(patient, facilityId);
+    const orgName = organization.data.name;
+    const orgOID = organization.oid;
+    const facilityNPI = facility.data["npi"] as string;
+
+    // Patients of cxs that not go through EC should have theis status undefined so they're not picked up later
+    // when we enable it
+    const cqLinkStatus = (await isEnhancedCoverageEnabledForCx(patient.cxId))
+      ? "unlinked"
+      : undefined;
+    const storeIds = getStoreIdsFn(patient.id, patient.cxId, cqLinkStatus);
+
+    commonWell = makeCommonWellAPI(orgName, oid(orgOID));
+
+    const commonwellData = getCWData(patient.data.externalData);
+
+    if (!commonwellData) {
+      const msg = `The patient does not have external data`;
+      debug(
+        `ERR - ${msg} - Patient not registered to commonwell - ` +
+          `respUpdate: ${JSON.stringify(commonwellData)}`
+      );
+      throw new Error(msg);
+    }
```

Comment:
I thought we were trying to link if not linked yet? So in theory shouldn't fail if this is not set.

Reply:
The way I conceptualized the problem is that the linking is either completed, processing, or failed. In all of those cases, the externalData does exist. If there is no externalData, then that means that the patient registration was never even attempted, which would mean create was never even called. 

This is a theoretically possible edge condition if `create` [fails](https://github.com/metriport/metriport/blob/79f7a1b886a77a2c1378b03d4b9a9f57d400d1ed/packages/api/src/external/commonwell/patient.ts#L95) between lines 95 and 124. I am adding a `[condition](https://github.com/metriport/metriport/blob/33f04b1a97d4e36bbdc27ac906fb90ba373db147/packages/api/src/external/commonwell/patient-external-data.ts#L43C1-L43C1)` in the gateway function to the rety linking logic to recreate/reregister the patient if there is no external data at all: